quarkus/tests/integration/src/test/java/org/keycloak/it/storage/database/MariaDBStartDatabaseTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.it.storage.database;
+
+import org.junit.jupiter.api.Test;
+import org.keycloak.it.junit5.extension.CLIResult;
+import org.keycloak.it.junit5.extension.CLITest;
+import org.keycloak.it.junit5.extension.WithDatabase;
+
+import io.quarkus.test.junit.main.Launch;
+import io.quarkus.test.junit.main.LaunchResult;
+
+@CLITest
+@WithDatabase(alias = "mariadb")
+public class MariaDBStartDatabaseTest {
+
+    @Test
```

Comment:
The standard tests are mostly copy-pasted across implementation, can we refactor into an `abstract class`?

Reply:
Not sure, they have specific messages accordingly to the database.

Reply:
I see, maybe you can parametrize the error message?

Reply:
I do not think it is worthy to introduce a hierarchy for these tests in particular. There is also the case for MSSQL where the `testSuccessful` requires some additional CLI args so we need to override in the sub-type.

But if you think this is a must I'm going to introduce the hierarchy as you are proposing.

Reply:
Done